### PR TITLE
fix: Disable create-reporting-task for NiFi 2.x and document regression

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by adwk67** [comment]
Currently testing this PR on openshift and `test_nifi_metrics.py` in e.g. the upgrade test doesn't detect any metrics.

**Review by sbernauer — COMMENTED** [review]
LGTM when tests pass

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ All notable changes to this project will be documented in this file.
   - `extraVolumes`
 - Increase `log` Volume size from 33 MiB to 500 MiB ([#671]).
 - Replaced experimental NiFi `2.0.0-M4` with `2.0.0` ([#702]).
+- Remove the automatic deployment of the PrometheusReportingTask for NiFi versions `2.x.x` and up ([#708]).
```
Comment (sbernauer) [inline CHANGELOG.md:24]:
```suggestion
- Don't deploy the `PrometheusReportingTask` Job for NiFi versions `2.x.x` and up ([#708]).
```

**Comment by sbernauer** [comment]
> Currently testing this PR on openshift and test_nifi_metrics.py in e.g. the upgrade test doesn't detect any metrics.

Good catch, the upgrade test probably also needs the same if clause

File: docs/modules/nifi/pages/usage_guide/monitoring.adoc [hunk 2]
```diff
@@ -3,8 +3,14 @@
 :k8s-job: https://kubernetes.io/docs/concepts/workloads/controllers/job/
 :k8s-network-policies: https://kubernetes.io/docs/concepts/services-networking/network-policies/
 
-The operator automatically configures NiFi to export Prometheus metrics.
-This is done by creating a {k8s-job}[Job] that connects to NiFi and configures a reporting task.
+In November 2024, Apache NiFi released a new major version https://cwiki.apache.org/confluence/display/NIFI/Release+Notes#ReleaseNotes-Version2.0.0[`2.0.0`].
+
+The NiFi `2.0.0` release changed the way of exposing Prometheus metrics significantly. The following steps explain on how to expose Metrics in NiFi versions `1.x.x` and `2.x.x`.
```
Comment (sbernauer) [inline docs/modules/nifi/pages/usage_guide/monitoring.adoc:8]:
```suggestion
The NiFi `2.0.0` release changed the way of exposing Prometheus metrics significantly.
The following steps explain on how to expose Metrics in NiFi versions `1.x.x` and `2.x.x`.
```

File: docs/modules/nifi/pages/usage_guide/monitoring.adoc [hunk 3]
```diff
@@ -24,3 +30,15 @@ spec:
     createReportingTaskJob:
       enabled: false
 ----
+
+== Configure metrics in NiFi `2.x.x`
+
+The Prometheus Reporting Task was removed in NiFi `2.x.x`. Metrics are now exposed automatically and can be scraped using the NiFi pod FQDN and the route `/nifi-api/flow/metrics/prometheus`.
```
Comment (sbernauer) [inline docs/modules/nifi/pages/usage_guide/monitoring.adoc:36]:
```suggestion
The Prometheus Reporting Task was removed in NiFi `2.x.x` in https://issues.apache.org/jira/browse/NIFI-13507[NIFI-13507].
Metrics are now always exposed and can be scraped using the NiFi Pod FQDN and the HTTP path `/nifi-api/flow/metrics/prometheus`.
```

File: docs/modules/nifi/pages/usage_guide/monitoring.adoc [hunk 4]
```diff
@@ -24,3 +30,15 @@ spec:
...
+
+For a deployed single node NiFi cluster called `simple-nifi`, containing a rolegroup called `default`, the metrics endpoint is reachable under:
+
+```
+https://simple-nifi-node-default-0.simple-nifi-node-default.<namespace>.svc.cluster.local:8443/nifi-api/flow/metrics/prometheus
+```
+
+IMPORTANT: If NiFi is configured to use any Authentication, requests to the metric endpoint must be authenticated and authorized.
```
Comment (sbernauer) [inline docs/modules/nifi/pages/usage_guide/monitoring.adoc:44]:
```suggestion
IMPORTANT: If NiFi is configured to do any user authentication, requests to the metric endpoint must be authenticated and authorized.
```

**Review by sbernauer — COMMENTED** [review]
Only small doc nits, LGTM otherwise (I ca not approve)

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by lfrancke** [comment]
Please include a link to the docs

**Comment by sbernauer** [comment]
Docs: https://docs.stackable.tech/home/nightly/nifi/usage_guide/monitoring/

**Pull request merged**